common/app/views/main.scala.html

```diff
@@ -27,34 +28,110 @@
     }
 }
 
-@immersiveMainMedia(page: GalleryPage) = {
-    <div class="new-gallery content tonal--tone-media immersive-main-media">
-        @defining(
-            page.item.content.trail.trailPicture.head
-        ) { picture =>
-            @fragments.image(
-                picture = picture,
-                classes = Seq("immersive-main-media__img"),
-                widths = MainMedia.immersive,
-                imageAltText = "",
-                isImmersiveMainMedia = true
-            )
+@immersiveMainMedia(page: ContentPage) = {
```

Comment:
There are some funky nested pattern matches in here, which I'm not sure if it's possible to avoid.

Also, I should move this out into a fragment.